cloud/db_cloud_impl.cc

```diff
@@ -405,6 +393,34 @@ Status DBCloudImpl::DoCheckpointToCloud(
     return st;
   }
 
+  // Copy MANIFEST and CLOUDMANIFEST sequentially only after copying all data
+  // files
+
+  // MANIFEST file
+  auto current_epoch = cenv->GetCloudManifest()->GetCurrentEpoch();
+  auto manifest_fname = ManifestFileWithEpoch("", current_epoch);
+  auto tmp_manifest_fname = manifest_fname + ".tmp";
+  auto fs = base_env->GetFileSystem();
+  st = CopyFile(fs.get(), GetName() + "/" + manifest_fname,
```

Comment:
Should we copy to this temporary file right after we `GetLiveFiles` but before the sst file copy?

Just in case MANIFEST file is updated when we copy the sst file, the CLOUDMANFIEST and MANIFEST we copy to cloud is still consistent?

If it's safe to assume that MANIFEST file will never get updated during checkpointing, then this might not be necessary

Reply:
Hmm actually good point. I do change the logic here a bit. Before the temporary file got copied before the sst files were uploaded and now it happens after. Yes I should copy the file first, thanks for catching this.